components/server/src/prometheus-metrics.ts

```diff
@@ -100,4 +100,16 @@ export function increaseMessagebusTopicReads(topic: string) {
     messagebusTopicReads.inc({
         topic,
     })
+}
+
+const workspaceStartHistogram = new prometheusClient.Histogram({
+    name: 'gitpod_server_workspace_start_duration_seconds',
```

Comment:
I reckon this name might be confusing, because
1. it does not reflect actual workspace startup time, but rather the time spent during a workspace start before it hits ws-manager/Kubernetes
2. it also includes image builds, hence mixes two modes of operation which will render the metric all but meaningless (you basically see two modes in the distribution, one without image builds and one with).

In addition to the name, the metric currently does not actually capture the user experience. When opening `gitpod.io/#some-context/url` we first run `createWorkspace` which - depending on the response times of the Git hoster - can take a few seconds to complete as well.

It makes perfect sense to measure the time we spend in server to create a workspace and issue the start request. Ideally though we'd don't include the image build if there is one, and to get the end-user perspective consider the workspace creation case, too.

For introducing the workspace start SLO I reckon we would get by with a simple counter vec, counting how many requests succeed. WDYT?

Reply:
> In addition to the name, the metric currently does not actually capture the user experience. When opening gitpod.io/#some-context/url we first run createWorkspace which - depending on the response times of the Git hoster - can take a few seconds to complete as well.

Thanks for this info! I didn't know about this `createWorkspace` part.

> It makes perfect sense to measure the time we spend in server to create a workspace and issue the start request. Ideally though we'd don't include the image build if there is one, and to get the end-user perspective consider the workspace creation case, too.

Yeah exactly, our current workspace startup time metric only measures time spent on ws-manager. For full user experience, we need to include all steps in between "User requests a new workspace" until "The IDE opened in the user's browser". I think that if in between these 2 moments, there is an image being built that the user needs to wait for, then it should be included in the measured duration.

> For introducing the workspace start SLO I reckon we would get by with a simple counter vec, counting how many requests succeed. WDYT?

To reduce initial complexity sounds like a good idea 🙂. I can create the dashboard based on this, and update in the future after we get the histogram working